core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/variable/index/IndexVariableListener.java

```diff
@@ -95,12 +90,6 @@ private void updateIndexes(InnerScoreDirector<Solution_, ?> scoreDirector, Objec
                 scoreDirector.beforeVariableChanged(shadowVariableDescriptor, element);
                 shadowVariableDescriptor.setValue(element, i);
                 scoreDirector.afterVariableChanged(shadowVariableDescriptor, element);
-            } else if (i != startIndex) {
-                // It is possible to quit early when an element with the expected index is encountered
-                // and **if it's not the moved element**. For example, when X is moved from Ann[3] to Beth[3],
-                // we need to start updating Beth's elements at index 3 where X already has the expected index, but quitting
-                // there would be incorrect because all the elements after X need their indexes incremented.
-                return;
             }
```

Comment:
Looks like we're removing an optimization.
Does that mean the optimization was incorrect?